executor/runtime/docker/docker.go

```diff
@@ -1315,11 +1315,18 @@ func (r *DockerRuntime) processEFSMounts(c runtimeTypes.Container) ([]efsMountIn
 			return nil, fmt.Errorf("Invalid EFS mount (read/write flag): %+v", configInfo)
 		}
 
-		if r.awsRegion == "" {
-			// We don't validate at client creation time, because we don't get this during testing.
-			return nil, errors.New("Could not retrieve EC2 region")
+		isRealEfsID := strings.HasPrefix(emi.efsFsID, "fs-")
+		if isRealEfsID {
+			if r.awsRegion == "" {
+				// We don't validate at client creation time, because we don't get this during testing.
+				return nil, errors.New("Could not retrieve EC2 region")
+			} // Get the remote IP. -- Is this really the best way how? Go doesn't have a simpler API for this?
+			emi.hostname = fmt.Sprintf("%s.efs.%s.amazonaws.com", emi.efsFsID, r.awsRegion)
```

Comment:
The hostname can begin with a fs-* I would say if the regexp matches:

`^fs-[0-9a-f]+$`

then it is an EFS.

Reply:
got it. ty